File: server/datasets/python_source-code/Multiple-E4/type3_boolflag_1_substring_search.py
def strStr(haystack, needle):
    if not needle: return 0
    
    for i in range(len(haystack) - len(needle) + 1):
        is_match = True
        for j in range(len(needle)):
            if is_match:
                if haystack[i + j] != needle[j]:
                    is_match = False
                elif j == len(needle) - 1:
                    return i
                    
    return -1

File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_e4/type3_boolflag_1_substring_search.java
public class Solution {
    public int strStr(String haystack, String needle) {
        if (needle == null || needle.length() == 0) return 0;
        
        for (int i = 0; i <= haystack.length() - needle.length(); i++) {
            boolean is_match = true;
            for (int j = 0; j < needle.length(); j++) {
                if (is_match) {
                    if (haystack.charAt(i + j) != needle.charAt(j)) {
                        is_match = false;
                    } else if (j == needle.length() - 1) {
                        return i;
                    }
                }
            }
        }
        return -1;
    }
}
